source/src/WebSocketServer.cc

```diff
@@ -0,0 +1,364 @@
+      if(findIter != m_serviceMap.end()) {
+        dqm_error("Couldn't create service '{0}' twice", name);
+        return nullptr; 
+      }
+      WsCommandHandler *service = new WsCommandHandler(this, name);
+      m_serviceMap.insert(ServiceMap::value_type(name, service));
+      return service;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsRequestHandler *WsServer::createRequestHandler(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter != m_serviceMap.end()) {
+        dqm_error("Couldn't create service '{0}' twice", name);
+        return nullptr; 
+      }
+      WsRequestHandler *service = new WsRequestHandler(this, name);
+      m_serviceMap.insert(ServiceMap::value_type(name, service));
+      return service;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsService *WsServer::findService(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter == m_serviceMap.end()) {
+        return nullptr;
+      }
+      if(findIter->second->type() != SERVICE_TYPE) {
+        return nullptr;
+      }
+      return dynamic_cast<WsService*>(findIter->second);
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsCommandHandler *WsServer::findCommandHandler(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter == m_serviceMap.end()) {
+        return nullptr;
+      }
+      if(findIter->second->type() != COMMAND_TYPE) {
+        return nullptr;
+      }
+      return dynamic_cast<WsCommandHandler*>(findIter->second);
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsRequestHandler *WsServer::findRequestHandler(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter == m_serviceMap.end()) {
+        return nullptr;
+      }
+      if(findIter->second->type() != RPC_TYPE) {
+        return nullptr;
+      }
+      return dynamic_cast<WsRequestHandler*>(findIter->second);
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void WsServer::start() {        
+      // Set logging settings
+      m_server.set_access_channels(websocketpp::log::alevel::all);
+      m_server.clear_access_channels(websocketpp::log::alevel::frame_payload);
+
+      // Initialize Asio
+      m_server.init_asio();
+
+      // Register our message handler
+      m_server.set_message_handler(std::bind(&WsServer::onMessage, this, _1, _2));
+      m_server.set_open_handler(std::bind(&WsServer::onOpen, this, _1));
+      m_server.set_close_handler(std::bind(&WsServer::onClose, this, _1));
+
+      // Listen on port 9002
```

Comment:
why 9002? ;)

Reply:
Why not dude !

Reply:
Just that the comment says 9002 but you are actually listening on `m_port`